jwst/exp_to_source/exp_to_source.py

```diff
@@ -40,8 +40,26 @@ def exp_to_source(inputs):
             log.debug(f'Copying source {slit.source_id}')
             result_slit = result[str(slit.source_id)]
             result_slit.exposures.append(slit)
+            # store values for later use (after merge_tree)
+            # these values are incorrectly getting overwritten by
+            # the top model.
+            slit_bunit = slit.meta.bunit_data
```

Comment:
All of the recursion errors in the regtest run are coming from the attempt to run ``blendheaders`` on combined products, which works its way through the meta data trees for each input that contributed to a combined output. Hence I'm guessing something is messed up in the meta data trees of the inputs and could be a result of the reassignments happening here. These 4 lines that are trying to save bits of meta data are really just taking a view into the meta data of the original `slit` model for a given input, which are then trying to be copied into the meta data tree of `result_slit` later on. I wonder if the use of a simple view of meta data tree is causing problems. Perhaps try modifying these 4 lines to make actual copies, instead of a view, i.e. `slit_bunit = slit.meta.bunit_data.copy()` and see if that helps. Just a wild guess at this point.

Reply:
I don't know that the meta tree has a copy() method, but I think `.instance` might work?